### PR TITLE
Add protractor rule

### PR DESCRIPTION
**Pull request by gregmagolan**
Brings in protractor_web_test & protractor_web_test_suite from angular/angular repo for publishing as @bazel/protractor npm package

### CODE REVIEW

**Comment by jbedard** [comment]
Is this a direct copy-paste atm? I haven't looked into this rule much myself, but this might be an opportunity to change the API, or at the very least expand it to include things such as https://github.com/angular/angular/pull/30245

**Comment by gregmagolan** [comment]
> Is this a direct copy-paste atm? I haven't looked into this rule much myself, but this might be an opportunity to change the API, or at the very least expand it to include things such as [angular/angular#30245](https://github.com/angular/angular/pull/30245)

Mostly copy & paste and some refactoring of file locations, paths, BUILD files, etc... to conform to rules_nodejs conventions.

The change in [angular/angular#30245](https://github.com/angular/angular/pull/30245). Can land that in a follow up PR right after to keep things somewhat organized.

**Comment by gregmagolan** [comment]
PR good to go. `@bazel/protractor` npm package will bring in the `protractor` npm package as a transitive dep so you don't need to specify it in your package.json explicitly anymore.

File: .circleci/config.yml [hunk 1]
```diff
@@ -20,6 +20,13 @@ var_4: &init_environment
       # Setup circleci environment
       ./.circleci/env.sh
 
+      # Install GTK+ graphical user interface (libgtk-3-0), advanced linux sound architecture (libasound2)
+      # and network security service libraries (libnss3) & X11 Screen Saver extension library (libssx1)
+      # which are dependendies of chrome & needed for karma & protractor headless chrome tests.
+      # This is a very small install with the whole init_environment step taking less than 8 seconds.
```
Comment (alexeagle) [inline .circleci/config.yml]: todo for using headless_shell to get away from this requirement?
will be a nuisance for our users too

File: examples/app/package.json [hunk 2]
```diff
@@ -1,18 +1,12 @@
...
-    "@types/jasmine": "2.8.2",
-    "@types/node": "7.0.18",
-    "concurrently": "^3.5.1",
-    "jasmine": "2.8.0",
-    "protractor": "^5.4.2",
+    "@types/jasmine": "3.3.15",
     "typescript": "2.7.x"
   },
   "scripts": {
-    "pretest": "bazel run @nodejs//:yarn && webdriver-manager update --standalone false --gecko false $CHROMEDRIVER_VERSION_ARG",
```
Comment (alexeagle) [inline examples/app/package.json]: yum :ice_cream:

**Pull request merged**